### PR TITLE
Update uses of logger to use formatted strings

### LINKED ISSUE #167: [Source Code]: Update engine logging instances
### What's Up?

Many uses of `FastJEngine#log/warning/etc` still make use of string concatenation -- these methods provide a useful formatting system, and should be updated as such.

### Code of Conduct

- [X] I agree to follow FastJ's Code of Conduct

### PR DESCRIPTION
**Pull request by YeffyCodeGit**
### Update uses of logger to use formatted strings
Resolves #167 

## Bug Fixes

- (#167) Update all the uses of logging functions to use formatted strings instead of string concatenation

### CODE REVIEW

File: src/main/java/tech/fastj/engine/FastJEngine.java [hunk 1]
```diff
@@ -702,7 +702,7 @@ private static void sync() {
             try {
                 TimeUnit.MILLISECONDS.sleep((long) ((endTime - currentTime) * 1000L));
             } catch (InterruptedException exception) {
-                FastJEngine.warning("Interrupted while syncing game frame rate: " + exception.getMessage() + Arrays.deepToString(exception.getStackTrace()));
+                FastJEngine.warning("Interrupted while syncing game frame rate: {}" + exception.getMessage() + Arrays.deepToString(exception.getStackTrace()));
```
Comment (lucasstarsz) [inline src/main/java/tech/fastj/engine/FastJEngine.java:705]: Seems like you've added the {} formatting, but the string concatenation is still present.

File: examples/java/tech/fastj/examples/bullethell/scripts/BulletMovement.java [hunk 2]
```diff
@@ -39,7 +39,7 @@ public void update(GameObject obj) {
 
         Pointf newTranslation = obj.getTranslation();
         if (newTranslation.equals(originalTranslation)) {
-            FastJEngine.warning("not moving! bullet at " + originalTranslation + " with " + travelAngle + "f " + travelVector);
+            FastJEngine.warning("not moving! bullet at {} with {} f {}", originalTranslation, travelAngle, travelVector);
```
Comment (lucasstarsz) [inline examples/java/tech/fastj/examples/bullethell/scripts/BulletMovement.java:42]: There shouldn't be a space between the brackets and the `f`.

File: examples/java/tech/fastj/examples/bullethell/scripts/BulletMovement.java [hunk 3]
```diff
@@ -50,7 +50,7 @@ public void destroy() {
 
     public void bulletDied(GameObject obj) {
         FastJEngine.runAfterUpdate(() -> {
-            FastJEngine.log("death! of bullet " + travelAngle + "f " + travelVector);
+            FastJEngine.log("death! of bullet {} f {}", travelAngle, travelVector);
```
Comment (lucasstarsz) [inline examples/java/tech/fastj/examples/bullethell/scripts/BulletMovement.java:53]: As mentuoned above, there's an extra space between the first brackets and the `f`.

File: src/main/java/tech/fastj/engine/internals/ThreadFixer.java [hunk 4]
```diff
@@ -27,7 +27,7 @@ public static void start() {
                     try {
                         Thread.sleep(Long.MAX_VALUE);
                     } catch (InterruptedException exception) {
-                        FastJEngine.warning("Interrupted while in daemon sleep: " + exception.getMessage() + Arrays.deepToString(exception.getStackTrace()));
+                        FastJEngine.warning("Interrupted while in daemon sleep: {}" + exception.getMessage() + Arrays.deepToString(exception.getStackTrace()));
```
Comment (lucasstarsz) [inline src/main/java/tech/fastj/engine/internals/ThreadFixer.java:30]: Seems like you've added the {} formatting, but the string concatenation is still present here.

File: src/main/java/tech/fastj/systems/audio/MemoryAudioPlayer.java [hunk 5]
```diff
@@ -28,7 +28,7 @@ static void playAudio(MemoryAudio audio) {
         Clip clip = audio.getAudioSource();
 
         if (clip.isOpen()) {
-            FastJEngine.warning("Tried to play audio file \"" + audio.getAudioPath().toString() + "\", but it was already open (and likely being used elsewhere.)");
+            FastJEngine.warning("Tried to play audio file \"{}\",  but it was already open (and likely being used elsewhere.)", audio.getAudioPath().toString());
             return;
```
Comment (lucasstarsz) [inline src/main/java/tech/fastj/systems/audio/MemoryAudioPlayer.java:32]: Just an extra space before `but` that needs to be deleted.

**Review by lucasstarsz — CHANGES_REQUESTED** [review]
Thanks for making a PR to FastJ! It's always very appreciated 👍🏾 
Things are looking good so far, but there are a few changes I'd like you to make. Please review my comments below.

**Comment by YeffyCodeGit** [comment]
Thanks for the review, i'll fix the mentioned issues in a bit! 👍

**Comment by YeffyCodeGit** [comment]
Sorry, I didn't notice I had missing changes, but it should all be done now.

**Review by lucasstarsz — APPROVED** [review]
Looks good. Don't worry about that failed unit test -- that's to be fixed separately.

**Pull request merged**